src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -348,10 +408,18 @@ bool AOTCodeCache::Config::verify() const {
     log_debug(aot, codecache, init)("AOT Code Cache disabled: it was created with CompressedOops::shift() = %d vs current %d", _compressedOopShift, CompressedOops::shift());
     return false;
   }
+  if (_compressedOopBase != CompressedOops::base()) {
+    log_debug(aot, codecache, init)("AOT Code Cache disabled: it was created with CompressedOops::base() = %p vs current %p", _compressedOopBase, CompressedOops::base());
+    return false;
+  }
   if (_compressedKlassShift != (uint)CompressedKlassPointers::shift()) {
     log_debug(aot, codecache, init)("AOT Code Cache disabled: it was created with CompressedKlassPointers::shift() = %d vs current %d", _compressedKlassShift, CompressedKlassPointers::shift());
     return false;
   }
+  if (_compressedKlassBase != CompressedKlassPointers::base()) {
+    log_debug(aot, codecache, init)("AOT Code Cache disabled: it was created with CompressedKlassPointers::base() = %p vs current %p", _compressedKlassBase, CompressedKlassPointers::base());
+    return false;
+  }
```

Comment:
I would suggest to use relocation for klass's base too but not in these changes. I think bailout is fine here.